### PR TITLE
Show pull request info in "Current Branch"

### LINKED ISSUE #2820: Show pull request info in "Current Branch" toolbar button
See https://github.com/desktop/desktop/issues/2642#issuecomment-330295922 for designs and discussion.

### PR DESCRIPTION
**Pull request by joshaber**
Fixes #2820

Builds on top of #2823.

screen shot 2017-09-28 at 1 19 37 pm
screen shot 2017-09-28 at 1 19 42 pm

### CODE REVIEW

**Comment by joshaber** [comment]
This is ready but it builds on top of #2823 so go review that first 🙏

**Comment by joshaber** [comment]
@niik I'm using `$gray-800` for the badge background, as per https://github.com/desktop/desktop/issues/2642#issuecomment-330295922, but it looks less contrast-y than your mock. Just FYI.

**Comment by j-f1** [comment]
Does it look better with the same color as the “Current Branch” text?

**Comment by joshaber** [comment]
This is ready now that #2823 has landed ✈️

**Comment by niik** [comment]
> @niik FYI that I'm using $gray-800 for the badge background, as per #2642 (comment), but it looks less contrast-y than your mock.

Hmm, that's odd. Could you drop me a screenshot and I'll take a look?

File: app/styles/ui/_pull-request-badge.scss [hunk 1]
```diff
@@ -0,0 +1,28 @@
+.toolbar-dropdown.open .pr-badge {
+  background: $gray-200;
```
Comment (niik) [inline app/styles/ui/_pull-request-badge.scss]: I realize that I forgot to spec the open button. Could we please bump this to 300?

File: app/styles/ui/ci-status.scss [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+.ci-status-pending {
+  color: $orange-700;
+}
+
+.ci-status-failure {
+  color: $red-600;
+}
+
+.ci-status-success {
+  color: $green-500;
```
Comment (niik) [inline app/styles/ui/ci-status.scss]: Can we make this 700 when the button is open?

File: app/styles/ui/ci-status.scss [hunk 3]
```diff
@@ -0,0 +1,16 @@
+.ci-status {
+  width: 16px;
+  flex-shrink: 0;
+}
+
+.ci-status-pending {
+  color: $orange-700;
```
Comment (niik) [inline app/styles/ui/ci-status.scss]: Can we make this 800 when the button is open?

**Comment by joshaber** [comment]
Screenshots updated in the body 🐗

File: app/styles/ui/_pull-request-badge.scss [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+.pr-badge {
+  display: flex;
+  flex-direction: row;
+  height: 18px;
+
+  align-items: center;
+
+  border: var(--base-border);
```
Comment (niik) [inline app/styles/ui/_pull-request-badge.scss]: Do we still need this given that `border-width` is zero down below?

File: app/styles/ui/_pull-request-badge.scss [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  border-radius: var(--border-radius);
+  border-width: 0;
+
+  margin-right: var(--spacing);
+  margin-left: var(--spacing);
```
Comment (niik) [inline app/styles/ui/_pull-request-badge.scss]: The text container already has a 10px right margin, so with this we're getting 20px. Can we ditch this?

File: app/styles/ui/_pull-request-badge.scss [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  padding: var(--spacing-half);
+
+  .number {
+    font-size: var(--font-size-sm);
+    line-height: 12px;
```
Comment (niik) [inline app/styles/ui/_pull-request-badge.scss]: Can we make this 18px, i.e. the height of the container?

File: app/src/ui/toolbar/branch-dropdown.tsx [hunk 7]
```diff
@@ -128,23 +130,24 @@ export class BranchDropdown extends React.Component<
 
       const pullRequestsWithStatus: Array<IPullRequest> = []
       for (const pr of pullRequests) {
```
Comment (niik) [inline app/src/ui/toolbar/branch-dropdown.tsx]: Just a note here that it'd be real nice if we could parallelize the fetching of status a bit, maybe run 5 or so requests in parallel but I imagine that requires some more sophisticated queuing work and running them all in parallel seems excessive.

File: app/src/ui/toolbar/branch-dropdown.tsx [hunk 8]
```diff
@@ -170,6 +173,20 @@ export class BranchDropdown extends React.Component<
     this.setState({ pullRequests })
   }
 
+  private get currentPullRequest(): IPullRequest | null {
```
Comment (niik) [inline app/src/ui/toolbar/branch-dropdown.tsx]: For the future it'd be real nice if we could get this into state and only recalculate when necessary

File: app/src/ui/toolbar/branch-dropdown.tsx (hunk 7 above)
Comment (joshaber) [reply app/src/ui/toolbar/branch-dropdown.tsx]: Yeah and this is all just placeholder so we can present some real data.

File: app/src/ui/toolbar/branch-dropdown.tsx (hunk 8 above)
Comment (joshaber) [reply app/src/ui/toolbar/branch-dropdown.tsx]: Same as above, just placeholder.

**Comment by joshaber** [comment]
🇻🇦

File: app/styles/ui/_pull-request-badge.scss [hunk 9]
```diff
@@ -0,0 +1,47 @@
+.toolbar-dropdown.open .pr-badge {
+  background: $gray-300;
+}
+
+.toolbar-dropdown:not(.open) .pr-badge {
+  background: $gray-700;
+}
+
+.pr-badge {
+  --height: 18px;
```
Comment (niik) [inline app/styles/ui/_pull-request-badge.scss]: 😍

**Pull request merged**